superset-frontend/src/views/CRUD/data/query/QueryList.tsx

```diff
@@ -0,0 +1,377 @@
+    if (status === 'failed') return theme.colors.error.base;
+    if (status === 'running') return theme.colors.primary.base;
+    if (status === 'offline') return theme.colors.grayscale.light1;
+
+    return theme.colors.grayscale.base;
+  }};
+`;
+
+function QueryList({ addDangerToast, addSuccessToast }: QueryListProps) {
+  const {
+    state: { loading, resourceCount: queryCount, resourceCollection: queries },
+    fetchData,
+  } = useListViewResource<QueryObject>(
+    'query',
+    t('Query History'),
+    addDangerToast,
+    false,
+  );
+
+  const menuData: SubMenuProps = {
+    activeChild: 'Query History',
+    ...commonMenuData,
+  };
+
+  const initialSort = [{ id: 'changed_on', desc: true }];
+  const columns = useMemo(
+    () => [
+      {
+        Cell: ({
+          row: {
+            original: { status },
+          },
+        }: any) => {
+          const statusConfig = {
+            name: '',
+            label: '',
+            status: '',
+          };
+          if (status === 'success') {
+            statusConfig.name = 'check';
+            statusConfig.label = t('Success');
+            statusConfig.status = 'success';
+          }
+          if (status === 'failed' || status === 'stopped') {
+            statusConfig.name = 'x-small';
+            statusConfig.label = t('Failed');
+            statusConfig.status = 'failed';
+          }
+          if (status === 'running') {
+            statusConfig.name = 'running';
+            statusConfig.label = t('Running');
+            statusConfig.status = 'running';
+          }
+          if (status === 'timed_out') {
+            statusConfig.name = 'offline';
+            statusConfig.label = t('Offline');
+            statusConfig.status = 'offline';
+          }
+          if (status === 'scheduled' || status === 'pending') {
+            statusConfig.name = 'queued';
+            statusConfig.label = t('Scheduled');
+            statusConfig.status = 'queued';
+          }
+          return (
+            <Tooltip title={statusConfig.label} placement="bottom">
+              <StatusIcon
+                name={statusConfig.name as IconName}
+                status={statusConfig.status}
+              />
+            </Tooltip>
+          );
+        },
+        accessor: 'status',
+        size: 'xs',
+        disableSortBy: true,
+      },
+      {
+        accessor: 'start_time',
+        Header: t('Time'),
+        size: 'lg',
+        Cell: ({
+          row: {
+            original: { start_time, end_time },
+          },
+        }: any) => {
+          const startDate = new Date(start_time);
+          const startUtc = new Date(
+            Date.UTC(
+              startDate.getFullYear(),
+              startDate.getMonth(),
+              startDate.getDate(),
+              startDate.getHours(),
+              startDate.getMinutes(),
+              startDate.getSeconds(),
+              startDate.getMilliseconds(),
+            ),
+          );
```

Comment:
this feels way too complicated, but maybe this is the only way to do it? Perhaps we can use a library like `moment` to convert into UTC?

Reply:
Was the intention to convert the database timestamp to local timestamp? If so, it's obviously not working (at least for my postgres database):

<img src="https://user-images.githubusercontent.com/335541/98911911-39d80300-247a-11eb-92bd-b0101172d2ed.png" width="500" />

Just ran a query at 11:58pm PT and got 15:58 UTC-8:00 instead.

Reply:
Since the backend already returns timestamps as UNIX epoch time, you can probably get by without doing special treatment for timezones:

```diff
diff --git a/superset-frontend/src/views/CRUD/data/query/QueryList.tsx b/superset-frontend/src/views/CRUD/data/query/QueryList.tsx
index 220c1a14e..0a223a536 100644
--- a/superset-frontend/src/views/CRUD/data/query/QueryList.tsx
+++ b/superset-frontend/src/views/CRUD/data/query/QueryList.tsx
@@ -194,20 +194,7 @@ function QueryList({ addDangerToast, addSuccessToast }: QueryListProps) {
             original: { start_time, end_time },
           },
         }: any) => {
-          const startDate = new Date(start_time);
-          const startUtc = new Date(
-            Date.UTC(
-              startDate.getFullYear(),
-              startDate.getMonth(),
-              startDate.getDate(),
-              startDate.getHours(),
-              startDate.getMinutes(),
-              startDate.getSeconds(),
-              startDate.getMilliseconds(),
-            ),
-          );
-          const startMoment = moment(startUtc);
-          const formattedStartTimeData = startMoment
+          const formattedStartTimeData = moment(start_time)
             .format(DATETIME_WITH_TIME_ZONE)
             .split(' ');
 
@@ -217,35 +204,18 @@ function QueryList({ addDangerToast, addSuccessToast }: QueryListProps) {
               {formattedStartTimeData[1]}
             </>
           );
-          if (!end_time) {
-            return formattedStartTime;
-          }
-
-          const endDate = new Date(end_time);
-          const endUtc = new Date(
-            Date.UTC(
-              endDate.getFullYear(),
-              endDate.getMonth(),
-              endDate.getDate(),
-              endDate.getHours(),
-              endDate.getMinutes(),
-              endDate.getSeconds(),
-              endDate.getMilliseconds(),
-            ),
-          );
-
-          return (
+          return end_time ? (
             <Tooltip
               title={t(
                 'Duration: %s',
-                moment
-                  .utc(moment(endUtc).diff(startMoment))
-                  .format(TIME_WITH_MS),
+                moment.utc(end_time - start_time).format(TIME_WITH_MS),
               )}
               placement="bottom"
             >
               <span>{formattedStartTime}</span>
             </Tooltip>
+          ) : (
+            formattedStartTime
           );
         },
       },

```

I'm also not sure how useful is the timezone to the users here. Most people would assume it is local time if no timezone is specified.

Reply:
To be clear this logic currently exists elsewhere in the list views (eg, https://github.com/preset-io/incubator-superset/blob/542d2e3b06095ff90bba382bf0b05adc8e62f91a/superset-frontend/src/views/CRUD/data/savedquery/SavedQueryList.tsx#L290)

The problem, as I understand it, is the these times are being stored without timestamps. So to avoid confusion, we assume the server times are in UTC as is common for most production servers. So any time that is sent from the server without a time zone is assumed to be in utc, so we must parse it as such to display correct results on the client. In your example @ktmud this was done locally where your dev server and db are running on your local timezone. We have had several issues with this before (things looking correct in local but wrong when they're on prod) and I believe this is the agreed upon safest approach.

I did manage to reduce this logic to only use moment though. 

cc @graceguo-supercat @riahk @dpgaspar since they're a little more familiar with the problem than I am

Reply:
As far as time zone info being useful or not, the goal of this PR is to reskin the previous view. Since time zone info existed in the previous view I think we should keep it for now and raise this issue separately where we can achieve consensus on this decision.

Reply:
The backend returns the timestamp as unix epoch time, which by definition is timezone-less. Maybe different API endpoints handle it differently?  @villebro have you done anything in this regard before, too?

Reply:
either way, it looks like this is working the same way across both versions 

<img width="1679" alt="Screen Shot 2020-11-12 at 9 38 51 AM" src="https://user-images.githubusercontent.com/10255196/98988568-f0180880-24cb-11eb-8a52-af660dd4b9aa.png">
<img width="1663" alt="Screen Shot 2020-11-12 at 9 39 29 AM" src="https://user-images.githubusercontent.com/10255196/98988578-f3ab8f80-24cb-11eb-99ee-959842686f25.png">

Reply:
Yeah, using `moment.utc(..).local(..)` is probably safer in case the backend returns ISO timestamp strings without timezones.

Reply:
@ktmud actually, you are correct. These are unix epochs, which are always in utc (or # seconds from midnight utc...1970). The problem I outlined above only applies to text-based datetimes. I've updated this logic to use your suggestions. Thanks for catching this!